packages/react-components/react-nav-preview/src/components/NavCategoryItem/NavCategoryItem.types.ts

```diff
@@ -10,20 +10,23 @@ export type NavCategoryItemSlots = {
   /**
    * The root element
    */
-  root: Slot<'button'>;
+  root: NonNullable<Slot<'button'>>;
 
   /**
-   * The component to be used as button in heading
+   * Icon that renders before the content.
    */
-  // button: NonNullable<Slot<ARIAButtonSlotProps<'a'>>>;
+  icon?: Slot<'span'>;
+
+  /**
+   * Component children are placed in this slot
+   * Avoid using the `children` property in this slot in favour of Component children whenever possible.
+   */
+  content: NonNullable<Slot<'span'>>;
+
   /**
    * Expand icon slot rendered before (or after) children content in heading.
```

Comment:
I think this should mention how this is different from `icon`.
```suggestion
   * Icon that shows the expanded vs collapsed state of a `NavCategoryItem`. This icon is rotated 90 degrees rather than using two icons.
```

On top of that, I think the prop to show the icon before or after is missing. I would have something like
```ts
/**
 * Whether the `expandIcon` should be rendered before or after the content.
 */
expandIconPosition?: 'before' | 'after';
```